tests/distribution_test.go

```diff
@@ -172,9 +211,13 @@ func createWithProps(t *testing.T) {
 	// Check prop1Values and prop2Values
 	assert.Len(t, prop1Values, 1)
 	assert.Len(t, prop2Values, 2)
-	assert.Equal(t, "value1", prop1Values[0])
-	assert.Equal(t, "value2", prop2Values[0])
-	assert.Equal(t, "value3", prop2Values[1])
+	if len(prop1Values) == 1 {
```

Comment:
What is the reason for this len check? We do an asset of this len a few lines above.
Same goes for ```if len(prop2Values) == 2``` below.

Reply:
The assertion doesn't fail immediately, so the length check prevents a NPE